### DISCUSSION TITLE
Setting default raw_user_meta_data?

### DISCUSSION THREAD

**Discussion by denull0** [supabase/supabase#4388] [Questions]
Hey everyone,

I am currently trying to set default value for raw_user_meta_data but I am failing big time. I tried doing it via the UI but that didn't work. I tried using this command:

`ALTER TABLE ONLY auth.users ALTER COLUMN raw_user_meta_data SET DEFAULT '{"onboarding":{"step1": false, "step2": false, "step3": false, "completed": false}}';`

But this didn't workout either.

I tried using a function + trigger but for some reason the syntax is wrong.

`
update auth.users
set raw_user_meta_data = ('{"onboarding":{"step1": true, "step2": true, "step3": true, "completed": true}}')
where id = auth.uid()
`

I tried using auth.signUp() and pass the data there, but sadly signUp doesn't work with magic links.

Any suggestions?

Thanks!

**Comment by sasweb** [comment]
You can pass any data to `signUp` or `signInWithOtp` (magic link) which is stored as raw_user_meta_data:

```
const { data, error } = await apiClient.auth.signInWithOtp({
  email,
  options: {
    shouldCreateUser: true,
    data: {
      onboarding: {
        step1: true,
        step2: true,
        step3: true,
        completed: true,
      },
    },
  },
});
```

Maybe it helps.
